lib/trie/proof.go

```diff
@@ -100,9 +102,11 @@ func VerifyProof(proof [][]byte, root []byte, items []Pair) (bool, error) {
 
 	for _, item := range items {
 		recValue := proofTrie.Get(item.Key)
-
+		if recValue == nil {
+			return false, ErrKeyNotFound
+		}
 		// here we need to compare value only if the caller pass the value
-		if item.Value != nil && !bytes.Equal(item.Value, recValue) {
+		if len(item.Value) > 0 && !bytes.Equal(item.Value, recValue) {
```

Comment:
Do we need this len check if it's a byte slice?

This most likely suffices I think:

```suggestion
		if !bytes.Equal(item.Value, recValue) {
```

Reply:
In the substrate code base, the `item.Value` (aka expected value) is an `Option`, so if the value `is_some()` then they compare otherwise they just skip the comparison. Here we just skip the comparison if the `item.Value` is empty as we don't have any expected value

Reply:
That makes sense.

I thought `bytes.Equal` would already do a len check as a first if block, but it turns out it just compares the string values of both with might not be that efficient ugh.